lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -139,95 +150,275 @@
           }
         }
 '''
+import json
+
+from abc import ABCMeta, abstractmethod
+
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.basic import AnsibleModule, get_all_subclasses
+from ansible.module_utils.six import with_metaclass
+
+
+class TestFailed(Exception):
```

Comment:
Also, the naming is odd. It's too generic and refers to software testing at the first look, misleading for the reader.
I suggest something more connected to its purpose:
```suggestion
class PackageManagerMissing(RuntimeError):
```
(or: s/Missing/Unavailable/)